test/test_torch.py

```diff
@@ -8032,6 +8032,10 @@ def _test_gather(self, cast, test_bounds=True):
         actual = torch.gather(src, 2, idx)
         self.assertEqual(actual, expected, 0)
 
+        # Bool test case
+        t = torch.tensor([[False, True],[True, True]])
+        self.assertEqual(torch.gather(t, 1, torch.tensor([[0, 0], [1, 0]])), torch.tensor([[False, False], [True, True]]))
```

Comment:
Needs cuda test case

Reply:
Imo `cast` takes care of it

Reply:
Ah, wait, it is supposed to take care of it =) but it is missing

Reply:
Ah, didn't see that, thanks for pointing that out